flow/layers/transform_layer.cc

```diff
@@ -25,19 +27,48 @@ void TransformLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
 
 #if defined(OS_FUCHSIA)
 
-void TransformLayer::UpdateScene(SceneUpdateContext& context,
-                                 mozart::Node* container) {
-  auto node = mozart::Node::New();
-  node->content_transform = mozart::Transform::From(transform_);
-  UpdateSceneChildrenInsideNode(context, container, std::move(node));
+void TransformLayer::UpdateScene(mozart::client::Session& session,
+                                 SceneUpdateContext& context,
+                                 ContainerNode& container) {
+  mozart::client::EntityNode node(&session);
+
+  // TODO(chinmaygarde): The perspective and shear components in the matrix are
+  // not handled correctly.
```

Comment:
Maybe we should have an API that takes the mat4 and let the scene manager deal with decomposition.